### PR TITLE
Add useTrackedDataQuery to react/query subpath

### PR DESCRIPTION
**Pull request by mcintyre94**
#### Summary of Changes

This PR adds `useTrackedDataQuery`, the Tanstack Query shaped version of `useTrackedData`.

```tsx
function AccountBalance({ address }: { address: Address }) {
    const client = useClient<ClientWithRpc<GetBalanceApi> & ClientWithRpcSubscriptions<AccountNotificationsApi>>();
    const spec = useMemo(
        () => ({
            initialValueSource: client.rpc.getBalance(address),
            initialValueMapper: (lamports: bigint) => lamports,
            streamSource: client.rpcSubscriptions.accountNotifications(address),
            streamValueMapper: ({ lamports }: { lamports: bigint }) => lamports,
        }),
        [client, address],
    );
    const { data, error } = useTrackedDataQuery(['balance', address], spec);
    if (error) return <p>Failed to load.</p>;
    return <p>{data ? `${data.value} lamports at slot ${data.context.slot}` : 'Loading…'}</p>;
}
```

Similarly to `useRequestQuery` the third param is an `UseQueryOptions`, with our usual `getAbortSignal` factory to add a custom signal. It also disables when either key or source are null.

Note that like `useSubscriptionQuery`, under the hood this uses [`experimental_streamedQuery`](https://tanstack.com/query/latest/docs/reference/streamedQuery), which allows `useQuery` to work with a `queryFn` that streams an `AsyncIterable`. Unlike SWR this does not use a different subscription type, so we expose the same API as any other `useQuery`. Most notably this means that we have `refetch` which is missing from `useSubscriptionSWR`. The cache is maintained across calls to `refetch`, meaning the slot will not regress. 

We use the same bridge function as `useSubscriptionQuery` to bridge from our `ReactiveStreamStore` into an `AsyncIterable`, in the same way that we bridged to SWR's custom subscription interface.

### CODE REVIEW

**Comment by mcintyre94** [comment]
* **#1786** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1786?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1782** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1782?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1780** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1780?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1779** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1779?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1769** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1769?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1769?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1760** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1760?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1759** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1759?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1758** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1758?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1754** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1754?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1747** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1747?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1727** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1727?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1719** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1719?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1713** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1713?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1707** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1707?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1709](https://github.com/anza-xyz/kit/pull/1709) <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1709?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1740** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1740?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1708** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1706** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1702** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1702?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1678** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1677** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1677?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1663** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1663?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1662** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1662?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1619** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1624** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1624?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1614** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1614?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1618](https://github.com/anza-xyz/kit/pull/1618) <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1618?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1628** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1628?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1612** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1612?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1607** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1607?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1606** <a href="https://app.graphite.com/github/pr/anza-xyz/kit/1606?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by mcintyre94** [comment]
@trevor-cortex

File: packages/react/src/useReactiveStoreLifecycle.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export function useReactiveStoreLifecycle<TStore extends { reset(): void }>(
             }
             if (++churn.count > CHURN_WARNING_THRESHOLD && !churn.warned) {
                 churn.warned = true;
-                // eslint-disable-next-line no-console
+
```
Comment (trevor-cortex) [inline packages/react/src/useReactiveStoreLifecycle.ts:87]: Unrelated to this PR — removes the `eslint-disable no-console` comment but the `console.error(...)` call right below is unchanged. If lint config hasn't been updated to permit this, CI will fail. If lint config has been updated, this still doesn't belong in a `useTrackedDataQuery` PR. Suggest reverting the change here and shipping it separately (or, if it's intentional, calling it out in the description).

File: packages/react/src/__tests__/useSubscription-test.browser.tsx [hunk 2]
```diff
@@ -97,26 +97,26 @@ describe('useSubscription', () => {
...
         expect(result.current.data).toStrictEqual({ context: { slot: 99n }, value: { lamports: 5n } });
     });
 
     it('passes raw notifications through unchanged', async () => {
-        const sub = makeFakeSubscription<{ slot: bigint; parent: bigint; root: bigint }>();
+        const sub = makeFakeSubscription<{ parent: bigint; root: bigint, slot: bigint; }>();
         const { result } = renderHook(() => useSubscription(sub.source));
 
-        await act(async () => sub.publish({ parent: 9n, root: 8n, slot: 10n }));
+        await act(async () => await sub.publish({ parent: 9n, root: 8n, slot: 10n }));
```
Comment (trevor-cortex) [inline packages/react/src/__tests__/useSubscription-test.browser.tsx:108]: Nit: this object type mixes separators — `parent: bigint; root: bigint, slot: bigint;` (note the `,` between `root` and `slot`). Either all `;` or all `,`. Also: this whole file's drive-by edits (`async () => sub.publish(...)` → `async () => await sub.publish(...)`, the `Notification<T>` reorder, this key reorder) are unrelated to `useTrackedDataQuery` and would be easier to land separately.

**Review by trevor-cortex — COMMENTED** [review]
## Summary

Adds `useTrackedDataQuery`, the TanStack-Query–shaped counterpart to `useTrackedData`. Pairs a one-shot RPC fetch with a long-lived subscription and routes the unified stream through TanStack's cache via `experimental_streamedQuery`, mirroring `useSubscriptionQuery`. Adds one genuinely novel piece on top: a cache-level slot floor (implemented as a `shouldYield` gate threaded into `bridgeStoreToAsyncIterable`) so a fresh per-attempt store's reconnect can't regress the cached envelope to an older slot from a lagging RPC node.

Overall this looks solid. The hook closely follows the `useSubscriptionQuery` shape (good for consistency), the slot-floor logic is the right mechanism for the cross-store dedup gap that the JSDoc explains well, and the tests cover the important shapes — surface, slot dedupe, null spec, sharing, teardown, abort wiring, and the cross-reconnect floor under `refetchMode: 'append'`. Docblocks and README are thorough. Type tests are comprehensive.

## Things worth a closer look

- **Unrelated change in `packages/react/src/useReactiveStoreLifecycle.ts`** — the `// eslint-disable-next-line no-console` comment is removed but the `console.error` call is unchanged. If lint config hasn't been updated to permit `no-console` here, CI will fail; if it has, this looks like accidental stray noise. Either way it doesn't belong in this PR. See inline.
- **Drive-by edits in `useSubscription-test.browser.tsx` and `useSubscriptionSWR-test.browser.tsx`** — purely cosmetic (`async () => sub.publish(...)` → `async () => await sub.publish(...)`, reordering of a union type, and a key reorder on a test object that introduces a stray `,`/`;` mix). Functionally fine, but they obscure the real diff and arguably belong in a separate cleanup PR. See inline for the stray-mixed-separator nit.
- **Slot-floor comparison uses `>=`** — that's correct (equal-slot envelopes carry the same logical state, and writing one isn't a regression) but means an equal-slot value will still write to the cache. Almost certainly fine in practice since the store's own internal slot tracker also dedupes, but worth confirming you're happy with the extra writes when a re-fetched same-slot envelope lands. Not a blocker.
- **`shouldYield` reads the cache on every emission via `queryClient.getQueryData(queryKey)`** — that's an in-memory hash lookup, so cheap, but it does mean every notification incurs an extra map read. Worth noting; not a blocker.

## Notes for subsequent reviewers

- The whole story hinges on `refetchMode: 'append'` keeping the prior envelope cached across a reconnect *and* `shouldYield` reading that cached envelope as the floor. Worth re-reading the "slot dedupe spans the whole cache" paragraph in the JSDoc and the `does not let an older-slot fetch on reconnect regress the warmer cached value` test together — they're the contract.
- The bridge's new `shouldYield` parameter is optional and `useSubscriptionQuery` (the other consumer) doesn't pass it, so behavior there is unchanged.
- The hook uses the same `experimental_streamedQuery<Envelope, Envelope>` + replace-latest-reducer trick as `useSubscriptionQuery`, including the `undefined as unknown as SolanaRpcResponse<TItem>` seed. Hacky but precedented.

File: packages/react/src/query/useTrackedDataQuery.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                    });
+                },
+            })(context),
+        [getAbortSignalRef, queryClient, specRef],
+    );
+
+    return useQuery<SolanaRpcResponse<TItem>, TError, TData, QueryKey>({
+        // Sensible defaults for a long-lived stream; all overridable by `queryOptions` below.
+        refetchOnWindowFocus: false,
+        retry: false,
```
Comment (trevor-cortex) [inline packages/react/src/query/useTrackedDataQuery.ts:171]: Slot-floor logic reads cleanly. One small thought: `>=` means an equal-slot envelope still passes the gate and writes to the cache. The store's own internal dedup likely makes this a non-issue in practice (it monotonically gates within a single store, so equal-slot re-emissions shouldn't really happen on the stream side), but across the cross-store boundary the cache could see a same-slot write from a fresh store after a reconnect. If you want to be strict, `>` would suppress those too. Up to you — `>=` is the safer default if there's any worry about strict-greater dropping a legitimate equal-slot promotion.

File: packages/react/src/query/useTrackedDataQuery.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            // The explicit `<Envelope, Envelope>` plus a replace-latest reducer types the cache value
+            // as the latest scalar envelope rather than an array. `initialValue` only seeds the
+            // (ignored) reducer accumulator before the first chunk, so its value is irrelevant — only
+            // its type matters.
+            experimental_streamedQuery<SolanaRpcResponse<TItem>, SolanaRpcResponse<TItem>>({
+                initialValue: undefined as unknown as SolanaRpcResponse<TItem>,
+                reducer: (_previous, chunk) => chunk,
+                // `'append'` is the only mode that writes every chunk to the cache live while leaving
+                // the prior value in place across a refetch. `'reset'` would flash to pending on each
+                // reconnect; `'replace'` only flushes once at stream end — which for a never-ending
```
Comment (trevor-cortex) [inline packages/react/src/query/useTrackedDataQuery.ts:140]: Same `undefined as unknown as SolanaRpcResponse<TItem>` seed pattern as `useSubscriptionQuery` (which uses `undefined as T`). For consistency with the sibling, consider `undefined as unknown as SolanaRpcResponse<TItem>` → just `undefined as SolanaRpcResponse<TItem>` (matches the sibling's casting style). Purely cosmetic; not a blocker.

File: packages/react/src/query/__tests__/useTrackedDataQuery-test.browser.tsx [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+
+import { renderHook } from '../../__test-utils__/render';
+import { TrackedDataSpec } from '../../useTrackedData';
+import { useTrackedDataQuery } from '../useTrackedDataQuery';
+
+// A fresh `QueryClient` per wrapper so cache state never leaks between tests, with the focus /
+// reconnect refetch triggers off so the test environment never re-fires queries mid-assertion.
+// `gcTime` defaults to 0 so an unmounted query is collected immediately; bump it for the cross-mount
+// slot-floor test that needs the cache to survive an unmount.
+function makeWrapper(gcTime = 0) {
```
Comment (trevor-cortex) [inline packages/react/src/query/__tests__/useTrackedDataQuery-test.browser.tsx]: The `gcTime = 0` default plus the comment hint that some test was originally going to need the cache to survive across an unmount, but the final test file doesn't seem to actually pass a non-zero `gcTime` anywhere (the cross-reconnect test stays mounted and uses `rerender` + `refetch`). If no test needs the bump anymore, consider dropping the parameter to keep the wrapper API minimal. Minor.

**Review by lorisleiva — APPROVED** [review]
Nice!

**Comment by mcintyre94** [comment]
### Merge activity

* **Jun 26, 1:11 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769).
* **Jun 26, 1:11 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) couldn't merge this pull request because a downstack PR #1607 failed to merge.
* **Jun 26, 1:39 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769).
* **Jun 26, 2:00 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) couldn't merge this pull request because a downstack PR #1628 failed to merge.
* **Jun 26, 2:17 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769).
* **Jun 26, 3:06 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) couldn't merge this pull request because a downstack PR #1663 failed to merge.
* **Jun 26, 3:26 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769).
* **Jun 26, 4:22 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) couldn't merge this pull request because a downstack PR #1708 failed to merge.
* **Jun 26, 6:12 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769).
* **Jun 26, 8:14 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) rebased this pull request as part of a merge.
* **Jun 26, 8:14 PM UTC**: [Graphite](https://app.graphite.com/github/pr/anza-xyz/kit/1769) couldn't merge this PR because it was not satisfying all requirements (Failed CI: 'Build & Test on Node current', 'Build & Test on Node lts/*').
